eclair-core/src/main/scala/fr/acinq/eclair/wire/LightningMessageCodecs.scala

```diff
@@ -16,22 +16,38 @@
 
 package fr.acinq.eclair.wire
 
-import fr.acinq.eclair.{KamonExt, wire}
 import fr.acinq.eclair.wire.CommonCodecs._
+import fr.acinq.eclair.{KamonExt, wire}
 import kamon.Kamon
 import kamon.tag.TagSet
-import scodec.bits.BitVector
-import scodec.{Attempt, Codec}
+import scodec.bits.{BitVector, ByteVector}
 import scodec.codecs._
+import scodec.{Attempt, Codec}
+import shapeless.HNil
 
 /**
  * Created by PM on 15/11/2016.
  */
 object LightningMessageCodecs {
 
-  val initCodec: Codec[Init] = (
-    ("globalFeatures" | varsizebinarydata) ::
-      ("localFeatures" | varsizebinarydata)).as[Init]
+  /** For historical reasons, features are divided into two feature bitmasks. We only send from the second one, but we allow receiving in both. */
+  val combinedFeaturesCodec: Codec[ByteVector] = (("globalFeatures" | varsizebinarydata) :: ("localFeatures" | varsizebinarydata)).exmap[ByteVector](
+    features => {
+      val f1 = features.head
+      val f2 = features.tail.head
+      val combinedFeatures = if (f1.length == f2.length) {
```

Comment:
nitpick: instead of creating a codec for a list and implicitly using the fact that it contains 2 elements, we could use a codec for a `(ByteVector, ByteVector)` instead and make this explicit, something like:
```scala
val combinedFeaturesCodec: Codec[ByteVector] = (("globalFeatures" | varsizebinarydata) :: ("localFeatures" | varsizebinarydata)).as[(ByteVector, ByteVector)].xmap(
    { case (f1, f2) =>
      val combinedFeatures = if (f1.length == f2.length) {
        f1 | f2
      } else if (f1.length > f2.length) {
        f1 | f2.padLeft(f1.length)
      } else {
        f1.padLeft(f2.length) | f2
      }
      combinedFeatures
    },
    { features => (ByteVector.empty, features) }
  )
```

Reply:
```scala
  val combinedFeaturesCodec: Codec[ByteVector] = (("globalFeatures" | varsizebinarydata) :: ("localFeatures" | varsizebinarydata)).as[(ByteVector, ByteVector)].xmap(
    { case (f1, f2) =>
      val length = f1.size.max(f2.size)
      f1.padLeft(length) | f2.padLeft(length)
    },
    { features => (ByteVector.empty, features) }
  )
```

Reply:
Thanks for the suggestions!